convokit/model/corpus.py

```diff
@@ -33,124 +41,227 @@ class Corpus:
         Type-checking ensures that the ConvoKitIndex is initialized correctly. However, it may be unnecessary if the
         index.json is already accurate and disabling it will allow for a faster corpus load. This parameter is set to
         True by default, i.e. type-checking is not carried out.
+    :param storage_type: either 'mem' or 'db'
+    :param storage: optional convokit.storage.StorageManager object to load the Corpus from
+    :param in_place: when connecting to a Corpus with data already in db storage, whether to load a copy of the Corpus 
+        or modify the existing Corpus in place.
+        True by default, i.e. Corpus is not initialized as a seperate copy.
+    :param from_corpus: if specified, the corpus to make a copy from.
+    :param version: (DB only): What version of the specified corpus to connect to
+    
 
     :ivar meta_index: index of Corpus metadata
     :ivar vectors: the vectors stored in the Corpus
     :ivar corpus_dirpath: path to the directory the corpus was loaded from
     """
-
-    def __init__(self, filename: Optional[str] = None, utterances: Optional[List[Utterance]] = None,
-                 preload_vectors: List[str] = None, utterance_start_index: int = None,
-                 utterance_end_index: int = None, merge_lines: bool = False,
+    def __init__(self,
+                 corpus_id: Optional[str] = None,
+                 utterances: Optional[List[Utterance]] = None,
+                 preload_vectors: List[str] = None,
+                 utterance_start_index: int = None,
+                 utterance_end_index: int = None,
+                 merge_lines: bool = False,
                  exclude_utterance_meta: Optional[List[str]] = None,
                  exclude_conversation_meta: Optional[List[str]] = None,
                  exclude_speaker_meta: Optional[List[str]] = None,
                  exclude_overall_meta: Optional[List[str]] = None,
-                 disable_type_check=True):
-
-        if filename is None:
-            self.corpus_dirpath = None
-        elif os.path.isdir(filename):
-            self.corpus_dirpath = filename
```

Comment:
BUG: The `corpus_dirpath` attribute has been deleted, but it's still being referenced in 10+ other places in the code. This means methods like `dump_vectors` and `load_info` are currently failing.

Reply:
Is this bug still an issue?

Reply:
I think `filename` is the new `corpus_dirpath`

Reply:
It looks like it's still being referenced in several parts of `corpus.py`

Reply:
yep, so what I did was change find `corpus_dirpath` and replace_all with `filename`, which in the param specifications is the same.

Reply:
Ah whoops, I thought you'd missed a few but guess not!